traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/router/TrafficRouter.java

```diff
@@ -559,9 +560,17 @@ public HTTPRouteResult multiRoute(final HTTPRequest request, final Track track)
 		return routeResult;
 	}
 
-	public String buildPatternBasedHashString(final DeliveryService deliveryService, final String requestPath) {
+	public String buildPatternBasedHashString(final DeliveryService deliveryService, final HTTPRequest request) {
+		final String requestPath = request.getPath();
 		if (deliveryService.getConsistentHashRegex() != null && !deliveryService.getConsistentHashRegex().isEmpty() && !requestPath.isEmpty()) {
-			return buildPatternBasedHashString(deliveryService.getConsistentHashRegex(), requestPath);
+			final StringBuilder hashString = new StringBuilder(request.getPath());
+			for (String qkey : request.getQueryString().split("&")) {
```

Comment:
I don't think so, but does that matter? shouldn't affect anything afaik.

Reply:
Well, we should probably figure that out. If it returned the decoded query string, then splitting on `&` won't work as expected if someone passes an encoded ampersand in the query string. There is also the question of what we should be hashing on -- decoded vs encoded? I'd think decoded, because `foo=%61` and `foo=a` should decode to the same thing, so we shouldn't hash them differently, right?

Reply:
I know for a fact that when initialized with a Java-standard `URL` object, it won't be decoded. I needed to check into that third party `HTTPRequest` to know what it does. It doesn't decode either.

> _"I'd think decoded..."_

I think that assumes that the origin is going to treat `%61` and `a` the same - which is probably true, maybe even _should_ be true, but doesn't _have_ to be true. I'm not sure if that's something we want to care about, though.

Reply:
>I think that assumes that the origin is going to treat %61 and a the same - which is probably true, maybe even should be true, but doesn't have to be true. 

I believe it is an RFC violation if it isn't.

https://tools.ietf.org/html/rfc3986#section-7.3
>Percent-encoded octets must be decoded at some point during the dereference process.

Reply:
That the origin needs to decode is enough reason for me.